### PR TITLE
vkd3d: Implement D3D12_HEAP_TYPE_WRITE_WATCH

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Needed for D3D12 APITrace

Closes: #373
Signed-off-by: Joshua Ashton <joshua@froggi.es>

### CODE REVIEW

**Comment by Joshua-Ashton** [comment]
Works under Windows, doesn't under wine because the `WRITE_WATCH` pointer fails to import.

I imagine there is some difference here in the way that Wine is emulating `WRITE_WATCH` that breaks somewhere down the line.

**Comment by Joshua-Ashton** [comment]
To test you can use `APITRACE_FORCE_MODULE_PATH=C:\Users\Joshua\Desktop\vkd3d-proton-write-watch\x64`, etc to point to your VKD3D-Proton + DXVK DXGI dlls for APITrace to use.

File: libs/vkd3d/memory.c [hunk 1]
```diff
@@ -299,6 +369,7 @@ static HRESULT vkd3d_memory_allocation_init(struct vkd3d_memory_allocation *allo
     VkMemoryAllocateFlagsInfo flags_info;
     VkMemoryPropertyFlags type_flags;
     uint32_t type_mask;
+    void *host_ptr = info->host_ptr;
```
Comment (HansKristian-Work) [inline libs/vkd3d/memory.c:372]: :christmas_tree:

File: libs/vkd3d/memory.c [hunk 2]
```diff
@@ -261,6 +261,73 @@ static HRESULT vkd3d_allocation_assign_gpu_address(struct vkd3d_memory_allocatio
     return S_OK;
 }
 
+static void *vkd3d_allocate_write_watch_pointer(const D3D12_HEAP_PROPERTIES* properties, VkDeviceSize size)
```
Comment (HansKristian-Work) [inline libs/vkd3d/memory.c:264]: `*` binds to right

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Nit, otherwise LGTM.

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**